### PR TITLE
Add Cursor based pagination support for Keys and Translations APIs

### PR DESCRIPTION
**Pull request by RonnyLV**
### Summary

Added new mode for paginated listing based on `cursor` on Keys and Translations endpoints.

### Other Information

Refer to [Pagination docs](https://developers.lokalise.com/reference/api-pagination) for more information

### CODE REVIEW

File: pagination.go [hunk 1]
```diff
@@ -8,16 +8,22 @@ import (
...
 }
 
 type Paged struct {
-	TotalCount int64 `json:"-"`
-	PageCount  int64 `json:"-"`
-	Limit      int64 `json:"-"`
-	Page       int64 `json:"-"`
+	TotalCount int64  `json:"-"`
+	PageCount  int64  `json:"-"`
+	Limit      int64  `json:"-"`
```
Comment (Dosexe) [inline pagination.go:24]: 🟢 So we are keeping page and limit for now + adding the cursor?

File: pagination.go [hunk 2]
```diff
@@ -28,13 +34,19 @@ func (p Paged) CurrentPage() int64 {
 	return p.Page
 }
 
+func (p Paged) NextCursor() string { return p.Cursor }
+
+func (p Paged) HasNextCursor() bool { return p.Cursor != "" }
```
Comment (Dosexe) [inline pagination.go:39]: 🟢 Maybe just `HasNextPage`? Usually I see this wording, however up to you what to use.

File: README.md [hunk 3]
```diff
@@ -101,6 +101,44 @@ t.SetPageOptions(lokalise.PageOptions{
 resp, err := t.List()
 ```
 
+### Cursor pagination
+
+The [List Keys](https://developers.lokalise.com/reference/list-all-keys) and [List Translations](https://developers.lokalise.com/reference/list-all-translations) endpoints support cursor pagination, which is recommended for its faster performance compared to traditional "offset" pagination. By default, "offset" pagination is used, so you must explicitly set `pagination` to `"cursor"` to use cursor pagination.
+
+```go
+// This approach is also applicable for `client.Translations()`
```
Comment (Dosexe) [inline README.md:109]: 🟢  You can add a link to `client.Translations()` here

**Review by Dosexe — APPROVED** [review]

File: pagination.go (hunk 1 above)
Comment (RonnyLV) [reply pagination.go:24]: Both modes are applicable and depends on the caller.

**Pull request merged**